ui/pages/settings/index.js

```diff
@@ -675,19 +675,138 @@ function Admins() {
   )
 }
 
-function Providers() {
+function Authentication() {
   const { data: { items: providers } = {} } = useSWR(
     `/api/providers?&limit=1000`
   )
+  const { data: org } = useSWR('/api/organizations/self')
+  const [allowedDomains, setAllowedDomains] = useState(
+    org?.allowedDomains || []
+  )
+  const [newDomain, setNewDomain] = useState('')
+  const [error, setError] = useState('')
+
+  async function removeDomain(e) {
+    e.preventDefault()
+    let toRemove = e.target.value
+    const newAllowedDomains = allowedDomains.filter(d => d !== toRemove)
+    updateAllowedDomains(newAllowedDomains)
+  }
+
+  async function addDomain(e) {
```

Comment:
Typing a non domain seems to work (e.g. `hi`). Should it? We may want to use a regex approach like [this](https://miguelmota.com/bytes/validate-domain-regex/), or validate this more server side

Reply:
Regex in our server-side validation doesn't work very well because it can't return a specific error message like other validations do, just that the pattern doesn't match. I opted to require a `.` for now, which is pretty permissive but should cover most invalid inputs.

Reply:
Sounds good. Did not mean we should use regex server side (it might be a good option client side though)